api/v1alpha1/clusterinstance_types.go

```diff
@@ -167,6 +181,10 @@ type NodeSpec struct {
 	// +required
 	HostName string `json:"hostName"`
 
+	// NodeRef is used to specify a reference to a BareMetalHost resource.
+	// +optional
+	NodeRef *NodeRef `json:"nodeRef,omitempty"`
```

Comment:
Let's use `HostRef` rather than `NodeRef` since this is indicating that we have a pre-existing BMH that we should use.